Thud Updates to Meta-Redhawk-SDR

michellely98 opened the pull request:
These are the changes to meta-redhawk-sdr that should allow it to run on Yocto 2.6 (thud) from 2.4 (rocko). Each change has a message explaining the reason.

***
A note: There is a 'patch' unlisted here that doesn't necessarily work as a patch, but shall be listed in this note anyway as to help troubleshoot a error that may pop up.

During the build, if " error: option --old-and-unmanageable not recognized" occurs and stops the build, try this:

in redhawk-codegen/setup.py, remove the line:

```
if not homeSys and ossiehome != None and not buildArg:
     sys.argv.append('--home='+ossiehome)
```

Then run the build. It will lead to another error. Add the above line back into the code, and run the build again. It should skip the 'old and unmanageable' error from there. Right now, this fix is unexplained on why it works. 
***

Commit:
edited layer.conf to change layercompatibility to 'thud'

Commit:
edited rh-sinkvita_49_3.1.0.bb to include patches

Commit:
edited rh-sourcevita49_3.1.0.bb recipe to include patches

Commit:
added PROVIDES to bulkiointerfaces recipe

Commit:
Added InPortImp.patch

Commit:
added python-setuptools to dependencies in attempt to solve redhawk-codegen error

Commit:
changed dependencies python-* to just python. Since ossp-uuid was depreciated in Yocto 2.6, all instances of of ossup-uuid were replaced by util-linux. Added patches for ThreadedComponent.cpp and and signalling.h

Commit:
add daemon_unixcc.patch

Commit:
pass a string in these cpp files as a string literal

Commit:
add patches for these cpp files in rh-sourcevita49

Commit:
include InPortImpl.patch to burstiointerfaces for variable casting

Commit:
Patch for ThreadedComponent.cpp that casts a posix_time number to unsigned long

Commit:
add patch to signalling.h to fix variable casting

Commit:
add daemon_unixcc patch to cast a string as a string literal

btgoodwin commented on conf/layer.conf:
```diff
@@ -28,14 +28,16 @@ BBFILE_COLLECTIONS += "redhawk-sdr"
     meta-python \
     "
-
+#PREFERRED_PROVIDER_python-threading ?= "bulkiointerfaces"
```
These 3 commented lines are unnecessary.  Please remove.

btgoodwin commented on recipes-core/components/rh-sinkvita49_3.1.0.bb:
```diff
@@ -2,7 +2,12 @@ DESCRIPTION = "REDHAWK SinkVITA49"
 DEPENDS = "bulkiointerfaces rh-vita49"
 RDEPENDS_${PN} = "bulkiointerfaces rh-vita49"
-
+LIC_FILES_CHSUM = "0"
```
This variable is invalid (and unnecessary since the checksum gets set in `recipes-core/core-framework/core-framework.inc`).  Please remove.

michellely98 replied:
Got it.

btgoodwin commented on recipes-core/core-framework/bulkiointerfaces_2.2.2.bb:
```diff
@@ -25,6 +25,11 @@ DESCRIPTION = "REDHAWK Framework BulkIO Interfaces"
 DEPENDS += "redhawk"
 RDEPENDS_${PN} = "redhawk"
 RDEPENDS_${PN}-python = "redhawk-python"
+#this was added to fix provide errors
```
None of these changes are necessary.  Please remove.

michellely98 replied:
I'll do this now.

btgoodwin commented on recipes-core/core-framework/redhawk-codegen_2.2.2.bb:
```diff
@@ -18,11 +18,11 @@
 #
 require core-framework-2.2.2.inc
-inherit setuptools redhawk-sysroot
+inherit setuptools redhawk-sysroot  
```
None of these changes are necessary.  The fix referenced in the PR can be simplified to no-exec of the configure task (replace do_configure's function with `do_configure[noexec] = "1"`).  The distutils class changed in a way that caused `setup.py` to run in a way it wasn't meant to run, hence the errors.  We don't actually need that step.

btgoodwin commented on recipes-core/core-framework/redhawk_2.2.2.bb:
```diff
@@ -23,20 +23,20 @@ require core-framework-autotools.inc
 DESCRIPTION = "REDHAWK Core Framework"
 
 PR = "4"
+PREFERRED_PROVIDER_virtual/python-threading ?= "bulkiointerfaces"
```
This is a non-functional change; please remove. (Bitbake packages can't modify one another in this way.)

btgoodwin reviewed (changes requested):
I've also sent you 4 patch files to apply (the pair for xsd replace the one patch that is there).  Please incorporate to remove nuisance warnings during `do_patch` tasks.

michellely98 committed:
Removed CHK_SUM from file

This variable is invalid (and unnecessary since the checksum gets set in recipes-core/core-framework/core-framework.inc).

michellely98 committed:
Removed Comments

michellely98 committed:
Removed Unnecessary Comments

michellely98 committed:
removed changes done in commit 

removed inherit and python-setuptools

michellely98 committed:
Update redhawk_2.2.2.b

Removed unnecessary Preferred Provider

Commit:
update fix_idldir patch from Thomas

Commit:
Update remove_csh_scripts patch from Thomas

Commit:
update xsd recipe to include patches from Thomas

Commit:
removed elements patch as its replaced by elements.hxx and elements.txx

Commit:
added elements.hxx and elements.txx patches

Commit:
Merge branch 'thud' of https://github.com/michellely98/meta-redhawk-sdr into thud

btgoodwin committed:
Removed python provides

This package does not provide any `python-*` packages.

btgoodwin committed:
Restored setuptools, redhawk-sysroot, removed configure

Previous commit removed required classes rather than the suggested configure task.  This restores those classes and noexec's the configure task.
